### DISCUSSION TITLE
Building hsdis for OpenJDK 15 - Gunnar Morling

### DISCUSSION THREAD

**Discussion by blog-comment-importer** [gunnarmorling/discussions.morling.dev#277] [Announcements]
https://www.morling.dev/blog/building-hsdis-for-openjdk-15/

**Comment by blog-comment-importer** [comment]
_**Stefan Reich** wrote at 2021-10-24T13:47:26Z (comment imported from Disqus):_

<p>Instructions not working... (first command)</p><p>root@c8:~/dev/jdk-15-with-hsdis# git clone git@github.com:openjdk/jdk.git<br>Cloning into 'jdk'...<br>The authenticity of host '<a href="http://github.com" rel="nofollow noopener" title="github.com">github.com</a> (140.82.121.3)' can't be established.<br>RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.<br>Are you sure you want to continue connecting (yes/no)? yes<br>Warning: Permanently added '<a href="http://github.com" rel="nofollow noopener" title="github.com">github.com</a>,140.82.121.3' (RSA) to the list of known hosts.<br>git@github.com: Permission denied (publickey).<br>fatal: Could not read from remote repository.</p>

**Reply by cjjavellana** [reply]
Try to use https 
```bash
git clone --depth 1 -b jdk-17+8 https://github.com/openjdk/jdk.git
```

**Comment by juancn** [comment]
For OpenJDK 11 (yeah, it's old) on a new-ish Xcode 16, you need to force an old C standard:
```
 export CFLAGS="-std=c89 -DSTDC_HEADERS -w" 
```
and get an older binutils:
```
wget https://ftp.gnu.org/gnu/binutils/binutils-2.19.1.tar.bz2
```

**Reply by vermosen** [reply]
I was in trouble to build exactly this version until I saw your message. I think there might be a small typo though as I had to set the c standard to -std=c99 instead to make it work (gcc13 / rhel8)
